File: python/src/loxtep/connect.py
"""Connect facade (MCP: loxtep_connect)."""

from __future__ import annotations

from dataclasses import dataclass

from .connectors import ConnectorsApi
from .templates import TemplatesApi


@dataclass(frozen=True)
class ConnectFacade:
    connectors: ConnectorsApi
    templates: TemplatesApi

File: nodejs/src/client/connect.ts
/**
 * Connect facade (MCP: loxtep_connect).
 * Delegates to connectors and templates APIs — no HTTP rewrite.
 */

import type { createConnectorsApi } from './connectors.js';
import type { createTemplatesApi } from './templates.js';

export interface ConnectFacadeDeps {
  connectors: ReturnType<typeof createConnectorsApi>;
  templates: ReturnType<typeof createTemplatesApi>;
}

export function createConnectFacade(deps: ConnectFacadeDeps): {
  connectors: ConnectFacadeDeps['connectors'];
  templates: ConnectFacadeDeps['templates'];
} {
  return {
    connectors: deps.connectors,
    templates: deps.templates,
  };
}

export type ConnectFacade = ReturnType<typeof createConnectFacade>;
